sipa/forms.py

```diff
@@ -72,7 +72,25 @@ def __call__(self, *args, **kwargs):
             *args, readonly=True, **kwargs)
 
 
-class ContactForm(FlaskForm):
+class SpamCheckField(StringField):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
+        c = kwargs.pop('class', '') or kwargs.pop('class_', '')
+        kwargs['class'] = u'%s %s' % ('honey', c)
+        kwargs['autocomplete'] = 'off'
+        return super().__call__(*args, **kwargs)
+
+
+class SpamProtectedForm(FlaskForm):
+    # Adds a honypot for bots to the form.
+    # This field must not be filled out to submit the form.
+    # We're using 'website' as the field-name since we won't give bots a hint.
+    website = SpamCheckField(label="", validators=[Length(0, 0, "You seem to like honey.")])
```

Comment:
can't we just use

`website = SpamCheckField(label="", validators=[Length(0, 0, "You seem to like honey.")], render_kw={"class":"honey", "autocomplete":"off"})`

here instead of the additional `SpamCheckField`?

Reply:
This would work for the autocomplete attribute but not for the class. classes provided by `render_kw` are overwritten by classes provided in the template (`form-control` is provided by the template for all fields).

Reply:
Ah okay. The PR looks good then AFAICS.